### PR TITLE
OTGR: Enable parents

### PR DESCRIPTION
**Pull request by mbtools**
Closes https://github.com/larshp/abapGit/issues/3788

@filak-sap please test since I don't have a system that includes PARENTS. 

https://github.com/abapGit-tests/OTGR seems to work fine but doesn't include parents. Could you pls add such case to https://github.com/abapGit-tests/OTGR_with_parents?

### CODE REVIEW

**Comment by filak-sap** [comment]
Will do!

File: src/objects/zcl_abapgit_object_otgr.clas.abap [hunk 1]
```diff
@@ -103,30 +105,49 @@ CLASS ZCL_ABAPGIT_OBJECT_OTGR IMPLEMENTATION.
...
+          IF sy-subrc = 0.
+            <lv_field> = cl_pak_wb_domains=>co_activation_state-inactive.
+          ENDIF.
+          ASSIGN COMPONENT 'OBJ_TYPE_GROUP' OF STRUCTURE <ls_parent> TO <lv_field>.
+          IF sy-subrc = 0.
+            " Removed in the method serialize.
+            <lv_field> = ms_item-obj_name.
+          ENDIF.
+        ENDLOOP.
+      CATCH cx_root.
```
Comment (filak-sap) [inline src/objects/zcl_abapgit_object_otgr.clas.abap:149]: Is this the best what we can do? Catching the top level exception and not spitting a word looks like an anti pattern, especially in such a huge code block.

Where do you expect the first occurrence of exception? In `CREATE DATA` or `io_xml->read`?

**Comment by filak-sap** [comment]
Here we go: https://github.com/abapGit-tests/OTGR_with_parents/pull/1

File: src/objects/zcl_abapgit_object_otgr.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/objects/zcl_abapgit_object_otgr.clas.abap:149]: If anything fails in this block, it means the feature is not supported by this release. I will try to reduce try catch and make it more specific.

Comment (filak-sap) [reply src/objects/zcl_abapgit_object_otgr.clas.abap:149]: That's the devil - what if the constructor of `CLS_TYGR_PARENT` start raising a new exception in the case of XYZ? Or `io_xml->read()`? We will never find out.

File: src/objects/zcl_abapgit_object_otgr.clas.abap [hunk 2]
```diff
@@ -247,15 +274,25 @@ CLASS ZCL_ABAPGIT_OBJECT_OTGR IMPLEMENTATION.
 *   so if someone adds support for them in future, there will be no format change.
     APPEND INITIAL LINE TO ls_otgr-texts ASSIGNING <ls_groupt>.
 
+    " Parents (cls_tygr_parent) does not exist in lower releases
+    TRY.
+        CREATE DATA lo_parents TYPE TABLE OF ('CLS_TYGR_PARENT').
+        ASSIGN lo_parents->* TO <lt_parents>.
+      CATCH cx_root.
```
Comment (filak-sap) [inline src/objects/zcl_abapgit_object_otgr.clas.abap:281]: ditto

File: src/objects/zcl_abapgit_object_otgr.clas.abap (hunk 1 above)
Comment (mbtools) [reply src/objects/zcl_abapgit_object_otgr.clas.abap]: fixed

**Comment by mbtools** [comment]
This should be better. There is no option in AG to influence the order of objects. Therefore, if parent is serialized before child, you might see inactive objects. 

Of course, we need to say that trying to pull OTGR with parents into a lower release that does not include this feature will lead to errors/diffs.

**Comment by larshp** [comment]
@filak-sap any additional comments?

**Comment by larshp** [comment]
@filak-sap

**Comment by filak-sap** [comment]
@mbtools  well done! Thank you!

**Review by filak-sap — APPROVED** [review]

**Comment by larshp** [comment]
thanks 👍

**Pull request merged**